inject_test.go

```diff
@@ -257,6 +270,30 @@ func TestInject(t *testing.T) {
 		assert.Nil(t, out.Result.T2, "T2 must be nil")
 		assert.True(t, gave1 == out.Result.T1, "T1 must match")
 	})
+
+	t.Run("FieldsCanBeOptional", func(t *testing.T) {
```

Comment:
There's already a test for private embedded fields: EmbeddedUnexportedField.

There's also a test for a nested fx.In struct, but not the exact case you mentioned. I'll add one.